application/recognition/image_recognition.py

```diff
@@ -51,22 +54,22 @@ def _call_google_api(cls, artifact):
 
     @classmethod
     def _extract_tags(cls, response):
-        """Private method to extract new tags from Google api response."""
-        tags = []
+        """Returns new label and text annotations extracted from Google api response body"""
+        label_annotations = []
+        text_annotations = []
         if 'responses' in response:
             res_one = response['responses'][0]
             if 'labelAnnotations' in res_one:
                 labels = res_one['labelAnnotations']
                 for label in labels:
-                    tags.append(label['description'])
+                    label_annotations.append(label['description'])
             if 'textAnnotations' in res_one:
                 texts = res_one['textAnnotations']
                 for text in texts:
-                    tags.append(text['description'])
-        return tags
+                    text_annotations.append(text['description'])
+        return label_annotations, text_annotations
 
     @classmethod
-    def add_tags_artifact(cls, artifact, new_tags):
-        existing_tags = artifact.tags or []
-        tags = existing_tags + new_tags
-        artifact.update({"tags": tags})
+    def add_tags_artifact(cls, artifact, label_annotations, text_annotations):
+        artifact.update({'label_annotations': label_annotations,
```

Comment:
Are you certain this works? Because previously you were given an ESArtifact here so these methods are proably not available for you at this point.

Reply:
` artifact.update({"tags": tags})` certainly works. Updating with LabelAnotations and TextAnotations should work here since I changed the update method of artifact (see artifact.py line 46).

Reply:
"should work" - this obviously means I have not tested enough. I will come back when that is tested^^